x-pack/plugins/uptime/server/lib/requests/search/monitor_summary_iterator.ts

```diff
@@ -117,17 +149,20 @@ export class MonitorGroupIterator {
       this.bufferPos = 0;
     }
 
-    const results = await this.chunkFetcher(this.queryContext, this.searchAfter, size);
+    const results = await this.chunkFetcher(this.queryContext, this.searchAfter, CHUNK_SIZE);
     // If we've hit the end of the stream searchAfter will be empty
-
-    results.monitorGroups.forEach((mig: MonitorGroups) => this.buffer.push(mig));
+    results.monitorSummaries.forEach((ms: MonitorSummary) => this.buffer.push(ms));
     if (results.searchAfter) {
       this.searchAfter = results.searchAfter;
     }
 
+    if (!results.searchAfter) {
```

Comment:
```suggestion
    if (!results.searchAfter || results.monitorSummaries.length < CHUNK_SIZE) {
```

Reply:
We can't really do this because the refine phase will reject some documents. So, we might get fewer results, but not actually be at the end.

Reply:
I added a comment about this to the code